server/src/main/kotlin/core/pokerhands/StraightFlush.kt

```diff
@@ -0,0 +1,28 @@
+package core.pokerhands
+
+import core.Card
+import core.CardRank
+
+class StraightFlush(cards: List<Card>) : PokerHand(HandRank.STRAIGHT_FLUSH, cards) {
+
+    override fun isValidHand(): Boolean = isStraightFlush(cards)
+
+    override fun compareWithinRank(other: PokerHand): Int {
+        val thisRanks = this.cards.map{ it.rank }.toSet()
+        val otherRanks = other.cards.map{ it.rank }.toSet()
+
+        val thisHighestRank = getHighestRank(thisRanks)
+        val otherHighestRank = getHighestRank(otherRanks)
+
+        return thisHighestRank.strength - otherHighestRank.strength
+    }
+
+    private fun getHighestRank(ranks: Set<CardRank>): CardRank {
+        val lowestStraight = (CardRank.TWO..CardRank.FIVE).toSet() + CardRank.ACE
+
+        if (ranks == lowestStraight)
+            return CardRank.FIVE
+        else
+            return ranks.maxBy { it.strength } !!
+    }
+}
```

Comment:
Maybe we should implement some automatic code formatting checker for kotlin (if it exists), like rubocop (https://github.com/rubocop-hq/rubocop). It would remind us to avoid simple mistakes like this

Reply:
(There is no line break at the end of the file)

Reply:
Check out this link, looks quite promising:
https://github.com/pinterest/ktlint
I will add proper plugin to pom.

Reply:
Nice

Reply:
You can now run 'mvn verify' to see all linter suggestions. Later I will add proper configuration file to allow automated formatting (altough keep in mind that it should be used with caution).